src/lsp/lp_lsp.ml

```diff
@@ -442,7 +474,6 @@ let main std log_file =
     LIO.log_error "[BT]" bt;
     F.pp_print_flush !LIO.debug_fmt ();
     flush_all ();
-    close_out debug_oc;
-    close_out lp_oc
```

Comment:
Thanks to keep `close_out lp_oc` in comment. Then, it should be possible to merge, even if a few things are not completely clear to me on the LSP server side.

Reply:
done

Reply:
Mainly, I replaced the channel associated with `lp-log.txt` with a buffer (`Lp_doc.lp_logger`). Then when the commands are evaluated they write logs to the buffer. I take the the contents of the buffer and store them along with the position of the command in `Lp_doc.t.logs`.